include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1049 @@
+
+    /** Appends the decoded characters to a container
+
+        This overload allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class T>
+    BOOST_URL_CXX20_CONSTEXPR
+    T&
+    append_to(T& str) const;
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if two strings are equal after percent decoding
+      */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /// @copydoc operator==
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /// @copydoc operator==
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator==(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) == 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if two strings are not equal after percent decoding
+      */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) != 0;
+    }
+
+    /// @copydoc operator!=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) != 0;
+    }
+
+    /// @copydoc operator!=
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator!=(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) != 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<(
+        decoded_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return lhs.compare(rhs) < 0;
+    }
+
+    /// @copydoc operator<
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<(
+        decoded_view lhs,
+        string_view rhs) noexcept
+    {
+        return lhs.compare(rhs) < 0;
+    }
+
+    /// @copydoc operator<
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<(
+        string_view lhs,
+        decoded_view rhs) noexcept
+    {
+        return rhs.compare(lhs) > 0;
+    }
+
+    /** Compares two strings
+
+        @param lhs A string to compare
+        @param rhs A string to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    friend BOOST_CXX14_CONSTEXPR
+    bool
+    operator<=(
+        decoded_view lhs,
```

Comment:
shouldn't these all be `decoded_view const&`?

Reply:
it is 32 bytes, I think is what matters. but since this is premature optimization, I shouldn't have mentioned it :)